crates/typst-library/src/compute/foundations.rs

```diff
@@ -82,6 +82,21 @@ pub fn panic(
     Err(msg)
 }
 
+/// Display: Warning
+/// Category: foundations
+#[func]
+pub fn warn(
+    message: Spanned<EcoString>,
+    /// An optional hint to display for the warning
+    #[named]
+    hint: Option<EcoString>,
+) -> SourceResult<Never> {
+    let Spanned { v: message, span } = message;
+    Err(Box::new(vec![
+        SourceDiagnostic::warning(span, message).with_hints(hint.into_iter())
+    ]))
```

Comment:
Would this result in control flow being exited, so that the second line never runs in the following?

```typst
{
    warn("something’s fishy")
    [#emph[This should still run!]]
}
```

Reply:
yes. this should use the tracer's warning functionality instead.

Reply:
Thanks, that was useful!